craft_parts/sources/base.py

```diff
@@ -106,6 +107,17 @@ def update(self):
         """
         raise errors.SourceUpdateUnsupported(self.__class__.__name__)
 
+    @classmethod
+    def _run_output(cls, command, **kwargs) -> str:
+        try:
+            return (
+                subprocess.check_output(command, **kwargs)
+                # .decode(sys.getfilesystemencoding())
```

Comment:
If `**kwargs` is passed to `subprocess.check_output`, pyright assumes that the output is `str` instead of `bytes`. This seems to be a bug in pyright. The extra argument is not being used anyway, so I'll remove it instead of ignoring type check here.

Reply:
I'm not following...

You say that if you do

```
            output = subprocess.check_output(command, text=True)
```

and handle output as Unicode, pyright will complain?

Reply:
If I do `output = subprocess.check_output(command, **kwargs)` pyright assumes output is `str`. I removed the extra argument because it's not used nor required by the rest of the code, and because `text=True` could eventually be passed there.

Reply:
Change to always use `text=True` and remove decoding, as discussed with Facundo.